### ISSUE TITLE
Wayland xdg-decoration support

### ISSUE THREAD

**Issue by pkerling** [xbmc/xbmc#15219]
## Roadmap or todo suggestion
**Detailed Description**

Kodi should support the unstable xdg-decoration Wayland protocol so it can use native window borders in windowed mode on Linux/Wayland

### Context

The custom borders are a fill-in and look fugly :-)

### Additional context, screenshots or links
Here are some relevant links or screenshots

https://github.com/wayland-project/wayland-protocols/blob/master/unstable/xdg-decoration/xdg-decoration-unstable-v1.xml

*Note: Team Kodi will consider this item however we will not make any promises if it will be included.*

**Comment by pkerling** [comment]
Alternative: https://gitlab.gnome.org/jadahl/libdecoration

**Comment by christianrauch** [comment]
If Kodi decides to use `libdecoration` for client-side decorations, I am happy to help with the porting. I am a main contributor to the project and implemented the libdecoration support for GLFW, SDL and Blender.

**Comment by yol** [comment]
@christianrauch I'd definitely be interested, can you summarize the current state of libdecoration? Is it API-stable?

**Comment by christianrauch** [comment]
> Can you summarize the current state of libdecoration? Is it API-stable?

It has not been officially deemed API stable yet. The implementations in GLFW, SLD and Blender are not upstreamed yet, and we are mostly looking for feedback from "real-world" applications and their needs/requirements. The API might change if required to support functionality requested by clients.

**Comment by lrusak** [comment]
I don't see any reason why we can't try it if we can make the dependency optional and the code changes flexible :smile_cat:

**Comment by Conan-Kudo** [comment]
You can control it with libdecor, including exposing a mode where you don't draw them.

**Comment by popcornmix** [comment]
This would be appreciated for RPiOS.
The decorations were deemed too ugly, so we've [hacked](https://github.com/popcornmix/xbmc/commit/45d3e2cca87c631aa9b0cd369bf168435a155b60) the internal decoration support out and enable decorations server side. e.g. for labwc:
```
<windowRules>
<windowRule identifier=“Kodi" serverDecoration=“yes" />
</windowRules>
```
Supporting this in a less hacky way would be good.

**Comment by Conan-Kudo** [comment]
[SDL's implementation](https://github.com/libsdl-org/SDL/blob/main/src/video/wayland/SDL_waylandwindow.c) might show a simple way to use libdecor.

**Comment by Danny3** [comment]
Supporting native window borders / decorations would be great as it would finally allow me to use the "Klassy" window decorations that I'm using for years on KDE Plasma and all KDE and even non-KDE, but still Qt programs support just fine and even a few popular non-Qt like Firefox and VLC supports just fine:
https://github.com/paulmcauley/klassy
The one that Kodi has in windowed mode are really bad, especially compared to Klassy and they look out of place.
If they cannot be replaced by default, we should at least have an option to be able to switch the the default / native / custom window decorations that we want, like Firefox and Chromium lets us choose.

**Comment by BBPSBB** [comment]
[4653b5eff7.patch](https://github.com/user-attachments/files/24360394/4653b5eff7.patch)

I am in no way a C++ developer. But managed to get it to use the server side decoration like this:

<img width="1704" height="987" alt="Image" src="https://github.com/user-attachments/assets/7682c92c-da88-4cbf-b88b-37d1fe337dd5" />


not sure how good it is, if there are edge cases where this does not work, etc  so I did not create a PR 🙈
